lib/blkdev.c

```diff
@@ -39,13 +40,15 @@
 
 static long
 blkdev_valid_offset (int fd, off_t offset) {
+	int r;
 	char ch;
 
 	if (lseek (fd, offset, 0) < 0)
 		return 0;
-	if (read (fd, &ch, 1) < 1)
-		return 0;
-	return 1;
+	r = read (fd, &ch, 1);
```

Comment:
Maybe we can use read_all() from include/all-io.h here. This function is able to restart the read if necessary.

Reply:
Yes, read_all sounds like a great idea. I have adjusted the commit. Thank you for pointing me to the function!